Tests/Kernel/TestMemoryDeviceMmap.cpp

```diff
@@ -9,19 +9,43 @@
 #include <assert.h>
 #include <errno.h>
 #include <fcntl.h>
+#include <inttypes.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <sys/mman.h>
+#include <sys/stat.h>
 #include <unistd.h>
 
+static u8 read_buffer[0x100000];
+
 static ALWAYS_INLINE bool mem_chunk(int fd, u64 base, u64 length)
 {
     u64 mmoffset = base % sysconf(_SC_PAGESIZE);
     void* mmp = mmap(NULL, mmoffset + length, PROT_READ, MAP_SHARED, fd, base - mmoffset);
     return mmp != MAP_FAILED;
 }
 
+enum class ReadResult {
+    SeekFailure,
+    ReadFailure,
+    ReadSuccess,
+};
+
+static ALWAYS_INLINE ReadResult read_chunk(int fd, u64 base, u64 length)
+{
+    VERIFY(length <= sizeof(read_buffer));
+    auto rs = lseek(fd, base, SEEK_SET);
+    if (rs < 0) {
+        fprintf(stderr, "Couldn't seek to offset %" PRIi64 " while verifying: %s\n", base, strerror(errno));
+        return ReadResult::SeekFailure;
+    }
+    if (read(fd, read_buffer, length) < 0) {
+        return ReadResult::ReadFailure;
+    }
+    return ReadResult::ReadSuccess;
+}
+
 TEST_CASE(test_memory_access_device_mmap)
```

Comment:
This test case is specifically for the mmap operation, so we should either rename it or split the read testing into a separate case.